### ISSUE TITLE
tuareg-mode integration (particularly merlin-mode setup) issues

### ISSUE THREAD

**Issue by geoffder** [ManasJayanth/esy-mode#8]
I am using `esy-mode` by loading in the `.el` in my doom config, which works fine for the major `reason-mode` (with hooks for `merlin-mode` and `esy-mode`). I have not added a hook for `esy-mode`, but have found it to be on (but dormant, as the `esy-mode` comments say it should be in a project that doesn't look like an Esy one) in ocaml buffers anyway. Seems like it is globally active by default now that I'm looking.

### System:
* Emacs 28.0.50 (Doom)
* Manjaro

### package.json
```json
{
  "name": "bonsai_revery",
  "version": "0.0.1",
  "description": "Bonsai Revery",
  "license": "MIT",
  "homepage": "https://github.com/osener/bonsai_revery",
  "esy": {
    "build": "dune build -p #{self.name}",
    "buildsInSource": "_build"
  },
  "scripts": {
    "format": "esy dune build @fmt --auto-promote",
    "run": "esy x examples"
  },
  "dependencies": {
    "@opam/bonsai": "v0.14.0",
    "revery": "*"
  },
  "devDependencies": {
    "ocaml": "~4.9.0",
    "@opam/dune": ">= 2.5.0",
    "revery": "revery-ui/revery#8e0479e",
    "@opam/merlin": "*"
  },
  "resolutions": {
    "esy-sdl2": "2.0.10008"
  }
}
```

Despite being "on" in an Esy project with a `package.json`, the `merlin-command` variable is not set (remains as default = "opam") when ocaml buffers are loaded, and merlin version mismatches result. 

### merlin error (cursor over a module)
```
Error from syntax checker ocaml-merlin: Wrong type argument: characterp, nil
eldoc error: (error merlin: [ ... ]/bonsai_revery/_build/default/src/.bonsai_revery.objs/byte/bonsai_revery.cmi
seems to be compiled with OCaml 4.09.
But this instance of Merlin handles OCaml 4.11.)
```

### describe-variable, after opening tuareg-mode ocaml buffer.
```
merlin-command is a variable defined in ‘merlin.el’.
Its value is ‘opam’
```

### describe-variable, after off/on `esy-mode`. 
```
merlin-command is a variable defined in ‘merlin.el’.
Its value is
"/home/geoff/.esy/3____________________________________________________________________/i/opam__s__merlin-opam__c__3.4.1-533ba774/bin/ocamlmerlin"
Original value was ‘opam’
```

## Workaround:
Turning off `esy-mode` then turning it back on does correctly set `merlin-command`. Subsequent refresh of `tuareg-mode` with `M-x tuareg-mode` gets things working.

**Comment by ManasJayanth** [comment]
Thanks for reporting this

**Comment by geoffder** [comment]
Pull request with fix here:
https://github.com/ManasJayanth/esy-mode/pull/9

**Comment by ManasJayanth** [comment]
Fixed by #9
